lib/Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php

```diff
@@ -67,15 +67,14 @@ public function __construct(array $params, $username, $password, array $driverOp
         $this->setDriverOptions($driverOptions);
 
         set_error_handler(function () {});
-
-        if ( ! $this->_conn->real_connect($params['host'], $username, $password, $dbname, $port, $socket, $flags)) {
+        try {
+            if ( ! $this->_conn->real_connect($params['host'], $username, $password, $dbname, $port, $socket, $flags)) {
+                throw new MysqliException($this->_conn->connect_error, @$this->_conn->sqlstate ?: 'HY000', $this->_conn->connect_errno);
```

Comment:
You're targeting `master` branch, which already have [dropped the support for PHP 5](https://github.com/doctrine/dbal/blob/master/composer.json#L15).